opal/class/opal_lifo.h

```diff
@@ -104,8 +104,47 @@ static inline bool opal_lifo_is_empty( opal_lifo_t* lifo )
     return (opal_list_item_t *) lifo->opal_lifo_head.data.item == &lifo->opal_lifo_ghost;
 }
 
+#if OPAL_HAVE_ATOMIC_LLSC_PTR
 
-#if OPAL_HAVE_ATOMIC_CMPSET_128
+static inline opal_list_item_t *opal_lifo_push_atomic (opal_lifo_t *lifo,
+                                                       opal_list_item_t *item)
+{
+    opal_list_item_t *next;
+
+    do {
+        item->opal_list_next = next = (opal_list_item_t *) opal_atomic_ll_ptr (&lifo->opal_lifo_head.data.item);
+        opal_atomic_wmb ();
+    } while (!opal_atomic_sc_ptr(&lifo->opal_lifo_head.data.item, item));
+
+    opal_atomic_wmb();
```

Comment:
A `wmb` shouldn't have any effect on the relative order of the push's `sc` and the pop's `ll`...

Barriers don't ensure "completion", they enforce relative ordering within a thread of execution as perceived by other threads.